### DISCUSSION TITLE
How to prevent cephcluster from being deleted by mistake? And the ceph cluster crash caused by mistakenly deleting cephcluster?

### DISCUSSION THREAD

**Discussion by zeaone** [rook/rook#11299] [Q&A] [answered]
How to prevent **cephcluster crd** from being deleted by mistake? And the ceph cluster crash caused by mistakenly deleting **cephcluster crd**?

**Comment by travisn** [comment, marked as answer]
Rook adds a finalizer to the CRs so they won't be accidentally deleted while there is still data in the cluster. If you do accidentally delete a CR, you can recover it with [these instructions](https://rook.io/docs/rook/latest/Troubleshooting/disaster-recovery/#restoring-crds-after-deletion).

**Reply by zeaone** [reply]
Is there a way to block some delete commands? Like `kubectl delete cephcluster rook-ceph -nrook-ceph`

**Reply by travisn** [reply]
No, the finalizer is the only deletion prevention method.

**Reply by zeaone** [reply]
Can it be achieved using `webhook`? like `ValidatingAdmissionWebhook`

**Reply by travisn** [reply]
We have a webhook, but the [delete method](https://github.com/rook/rook/blob/master/pkg/apis/ceph.rook.io/v1/cluster.go#L66) is not implemented. PRs welcome. Before code is changed, I'd suggest opening an issue for discussing the approach, how it can be disabled, if it's controlled by an operator env var, etc.

**Reply by zeaone** [reply]
ok, thanks😄

**Comment by mfordjody** [comment]


**Reply by zeaone** [reply]
Thanks for your answer, I'll try it
